espnet2/asr/transducer/beam_search_transducer.py

```diff
@@ -244,18 +261,58 @@ def default_beam_search(self, enc_out: torch.Tensor) -> List[Hypothesis]:
             nbest_hyps: N-best hypothesis.
 
         """
+        logging.info("decoder input length: " + str(enc_out.shape[0]))
         beam = min(self.beam_size, self.vocab_size)
         beam_k = min(beam, (self.vocab_size - 1))
 
         dec_state = self.decoder.init_state(1)
 
         kept_hyps = [Hypothesis(score=0.0, yseq=[self.blank_id], dec_state=dec_state)]
         cache = {}
+        cache_lm = {}
+
+        def make_lm_tokens(yseq: List[int]) -> torch.Tensor:
+            """Make LM tokens from a list of tokens
+
+            If the first token is <blank>, then replace it with <sos>.
+            Return token sequence tensor for LM scoring.
+
+            """
+            if len(yseq) and yseq[0] == self.blank_id:
+                return torch.LongTensor(
+                    [self.sos] + yseq[1:], device=self.decoder.device
+                )
+            else:
+                return torch.LongTensor(yseq, device=self.decoder.device)
```

Comment:
This fix is only intended for default beam search, is there a reason outside handling padding?

Also, yseq is always a list with `blank_id` as first symbol so we could directly pass :

```python3
torch.LongTensor([self.sos] + max_hyp.yseq[1:], device=self.decoder.device)
```

to `lm_score(...)`?

Reply:
> This fix is only intended for default beam search, is there a reason?

Sorry, thats my mistake. I will add it to other algorithms as well.
> Also, yseq is always a list with blank_id as first symbol so we could directly pass :
You're right. I think I was being conservative. If the first token is always `<blank>`, then we can just pass `torch.LongTensor([self.sos] + max_hyp.yseq[1:], device=self.decoder.device)` to `lm.score()` without any additional functions. I'll make this change.